### PR TITLE
fix issue 716

### PR DESCRIPTION
**Pull request by lemire**
fixes https://github.com/ada-url/ada/issues/716

### CODE REVIEW

**Review by bungle — APPROVED** [review]

**Comment by anonrig** [comment]
It seems web-platform tests are failing with this change

File: include/ada/url-inl.h [hunk 1]
```diff
@@ -217,7 +217,7 @@ ada_really_inline size_t url::parse_port(std::string_view view,
   ada_log("parse_port('", view, "') ", view.size());
   uint16_t parsed_port{};
   auto r = std::from_chars(view.data(), view.data() + view.size(), parsed_port);
-  if (r.ec == std::errc::result_out_of_range) {
+  if (r.ec != std::errc()) {
     ada_log("parse_port: std::errc::result_out_of_range");
```
Comment (anonrig) [inline include/ada/url-inl.h:221]: I think we also need the change this line

**Comment by lemire** [comment]
@anonrig My fix is not correct.

**Comment by lemire** [comment]
@anonrig Annoyingly, it seems that we have to check for negative values separately. :-/ Latest commit should be correct.

File: include/ada/url-inl.h [hunk 2]
```diff
@@ -215,10 +215,15 @@ inline void url::copy_scheme(const ada::url &u) {
 ada_really_inline size_t url::parse_port(std::string_view view,
                                          bool check_trailing_content) noexcept {
   ada_log("parse_port('", view, "') ", view.size());
+  if (!view.empty() && view[0] == '-') {
```
Comment (anonrig) [inline include/ada/url-inl.h:218]: Should we add `ada_unlikely` to here?

**Review by anonrig — APPROVED** [review]

**Pull request merged**